### ISSUE TITLE
Unlock by email feature

### ISSUE THREAD

**Issue by rarguelloF** [aarondl/authboss#198]
I wonder if you would be open to implement this feature in the lock module, similar to what [devise](https://github.com/plataformatec/devise/blob/master/lib/devise/models/lockable.rb#L20) does.

Please confirm me and I could work on a PR, adding a configurable option for unlock strategy and making it optional.

Thanks in advance!

**Comment by aarondl** [comment]
Absolutely. I'll work with you to code review it etc. If you need any help/advice just add comments to this issue.

One thing to note is no new modules should be using the side effect import method.
